scipy/interpolate/_cubic.py

```diff
@@ -520,11 +504,10 @@ def __init__(self, x, y, axis=0, *, method: Literal["akima", "makima"]="akima",
 
         if np.iscomplexobj(y):
             msg = ("`Akima1DInterpolator` only works with real values for `y`. "
-                   "Passing an array with a complex dtype for `y` is deprecated "
-                   "and will raise an error in SciPy 1.15.0. If you are trying to "
-                   "use the real components of the passed array, use `np.real` on "
-                   "the array before passing to `Akima1DInterpolator`.")
-            warnings.warn(msg, DeprecationWarning, stacklevel=2)
+                   "If you are trying to use the real components of the passed array, "
+                   "use `np.real` on the array before passing to "
+                   "`Akima1DInterpolator`.")
+            raise ValueError(msg)
```

Comment:
I changed the type of the error for consistency with what is raised if `x` is complex